modules/core/include/opencv2/core/hal/intrin_cpp.hpp

```diff
@@ -594,25 +594,25 @@ template<typename _Tp, int n> CV_INLINE v_reg<_Tp, n>& operator/=(v_reg<_Tp, n>&
 
 /** @brief Bitwise AND
 
-Only for integer types. */
+For all types.. */
 template<typename _Tp, int n> CV_INLINE v_reg<_Tp, n> operator&(const v_reg<_Tp, n>& a, const v_reg<_Tp, n>& b);
```

Comment:
masks should be used instead of trying to modify buts of floating point types.

Reply:
Problem is still here.
We don't really want to declare bitwise operations for floating-point numbers in universal API.
You assumed here `0` and `0xffff...ff` values **only**, but it is not assumed in the API itself (and has no correct interpretation/definition for other values like 2.0f).

---

> The bitwise logic operator of floating-pointis used in the OpenCV source code, like in this color_hsv

It is not used until this patch.
Regular CI build passes: http://pullrequest.opencv.org/buildbot/builders/master_etc-simd-emulator-lin64
Build with changed `modules/imgproc/src/color_hsv.simd.hpp` from this patch fails (`cmake ... -DOPENCV_EXTRA_FLAGS="-DCV_FORCE_SIMD128_CPP=1" -DCPU_BASELINE=SSE4_2 -DCPU_DISPATCH=`):

```
/home/alalek/projects/opencv/dev/modules/imgproc/src/color_hsv.simd.hpp: In member function ‘void cv::hal::cpu_baseline::{anonymous}::RGB2HSV_f::process(const cv::hal_EMULATOR_CPP::simd128_cpp::v_float32&, const cv::hal_EMULATOR_CPP::simd128_cpp::v_float32&, const cv::hal_EMULATOR_CPP::simd128_cpp::v_float32&, cv::hal_EMULATOR_CPP::simd128_cpp::v_float32&, cv::hal_EMULATOR_CPP::simd128_cpp::v_float32&, cv::hal_EMULATOR_CPP::simd128_cpp::v_float32&, float) const’:
/home/alalek/projects/opencv/dev/modules/imgproc/src/color_hsv.simd.hpp:298:53: error: no matching function for call to ‘v_and(cv::hal_EMULATOR_CPP::simd128_cpp::v_float32, cv::hal_EMULATOR_CPP::simd128_cpp::v_float32)’
  298 |         v_float32 v_res = v_select(v_r_eq_max, v_and(v_lt(v_g, v_b), vx_setall_f32(360.0f)),
      |                                                ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/alalek/projects/opencv/dev/modules/core/include/opencv2/core/hal/intrin.hpp:757:19: note: candidate: ‘cv::hal_EMULATOR_CPP::simd128_cpp::v_uint8 cv::hal_EMULATOR_CPP::simd128_cpp::v_and(const v_uint8&, const v_uint8&)’
  757 |     inline _Tpvec v_and(const _Tpvec& a, const _Tpvec& b) \
      |                   ^~~~~
/home/alalek/projects/opencv/dev/modules/core/include/opencv2/core/hal/intrin.hpp:757:19: note: in definition of macro ‘OPENCV_HAL_WRAP_BIN_OP_LOGIC’
  757 |     inline _Tpvec v_and(const _Tpvec& a, const _Tpvec& b) \
      |                   ^~~~~
...
```

Reply:
> You assumed here 0 and 0xffff...ff values only, but it is not assumed in the API itself

Thanks for your comment! I got the point and revert the modify of the API.